benchmarks/elementary_functions_experiments_benchmark.cpp

```diff
@@ -0,0 +1,195 @@
+  auto const& cos_x₀ = accurate_values.cos_x;
+  auto const h = x - x₀;
+  auto const h² = h * h;
+  auto const h³ = h² * h;
+  auto const cos_x₀_minus_h_sin_x₀ = TwoProductNegatedAdd(sin_x₀, h, cos_x₀);
+  return cos_x₀_minus_h_sin_x₀.value +
+         ((cos_x₀ * h² * CosPolynomial(h²) - sin_x₀ * h³ * SinPolynomial(h²)) +
+          cos_x₀_minus_h_sin_x₀.error);
+}
+
+// TODO(phl): This should use polynomials.
+
+template<Argument table_spacing>
+Value Implementation<table_spacing>::SinPolynomial(Argument const x) {
+  if constexpr (table_spacing == 2.0 / 256.0) {
+    // 71 bits.
+    return -0.166666666666666666666421797625 +
+           0.00833333057503280528178543245797 * x;
+  } else if constexpr (table_spacing == 2.0 / 1024.0) {
+    // 85 bits.
+    return -0.166666666666666666666666651721 +
+           0.00833333316093951937646271666739 * x;
+  }
+}
+
+template<Argument table_spacing>
+Value Implementation<table_spacing>::CosPolynomial(Argument const x) {
+  if constexpr (table_spacing == 2.0 / 256.0) {
+    // 77 bits.
+    return -0.499999999999999999999999974543 +
+           x * (0.0416666666666633318024480868405 -
+                0.00138888829905860875255146938745 * x);
+  } else if constexpr (table_spacing == 2.0 / 1024.0) {
+    // 72 bits.
+    return -0.499999999999999999999872434553 +
+           0.0416666654823785864634569932662 * x;
+  }
+}
+
+template<Argument table_spacing>
+void BM_EvaluateSinTableSpacing(benchmark::State& state) {
+  std::mt19937_64 random(42);
+  std::uniform_real_distribution<> uniformly_at(x_min, x_max);
+
+  Implementation<table_spacing> implementation;
+  implementation.Initialize();
+
+  Argument a[number_of_iterations];
+  for (std::int64_t i = 0; i < number_of_iterations; ++i) {
+    a[i] = uniformly_at(random);
+  }
+
+  Value v[number_of_iterations];
+  while (state.KeepRunningBatch(number_of_iterations)) {
+    for (std::int64_t i = 0; i < number_of_iterations; ++i) {
+      using namespace principia::quantities;
+      v[i] = implementation.Sin(a[i]);
+#if _DEBUG
+      // The implementation is not accurate, but let's check that it's not
+      // broken.
+      auto const absolute_error = Abs(v[i] - std::sin(a[i]));
+      CHECK_LT(absolute_error, 5.6e-17);
+#endif
+    }
+    benchmark::DoNotOptimize(v);
+  }
+}
+
+template<Argument table_spacing>
+void BM_EvaluateCosTableSpacing(benchmark::State& state) {
+  std::mt19937_64 random(42);
+  std::uniform_real_distribution<> uniformly_at(x_min, x_max);
+
+  Implementation<table_spacing> implementation;
+  implementation.Initialize();
+
+  Argument a[number_of_iterations];
+  for (std::int64_t i = 0; i < number_of_iterations; ++i) {
+    a[i] = uniformly_at(random);
+  }
+
+  Value v[number_of_iterations];
+  while (state.KeepRunningBatch(number_of_iterations)) {
+    for (std::int64_t i = 0; i < number_of_iterations; ++i) {
+      using namespace principia::quantities;
+      v[i] = implementation.Cos(a[i]);
```

Comment:
Should we also benchmark latency?

Reply:
A latency benchmark is going to quickly converge to a fixed point.  That's not useful, because once all the invocations load from the same table entry, the influence of the table size disappears.  We really want to touch all the table entries.